detoks/src/main/java/nl/tudelft/trustchain/detoks/gossiper/BootGossiper.kt

```diff
@@ -0,0 +1,72 @@
+package nl.tudelft.trustchain.detoks.gossiper
+
+import android.util.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import nl.tudelft.ipv8.Peer
+import nl.tudelft.ipv8.android.IPv8Android
+import nl.tudelft.trustchain.detoks.DeToksCommunity
+
+class BootGossiper(
+    override val delay: Long,
+    override val peers: Int
+) : Gossiper() {
+
+    var maxLoops = (30000/delay).toInt()
+
+    override fun startGossip(coroutineScope: CoroutineScope) {
+        coroutineScope.launch {
+            while (coroutineScope.isActive && running) {
```

Comment:
The running variable is changed in the function in the companion object, so this cannot be changed to maxLoops, since this cannot be set in the companion object (since it uses delay in its initialisation which is a parameter of the object).